assets/scripts/streets/StreetNameCanvas.jsx

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash'
+import React from 'react'
+import StreetName from './StreetName'
+import StreetMetaData from './StreetMetaData'
+import { getStreet } from './data_model'
+import ReactDOM from 'react-dom'
+
+export default class StreetNameCanvas extends React.Component {
+  constructor (props) {
+    super(props)
+
+    var street = getStreet()
+    this.state = {
+      classNames: [],
+      street: street
+    }
+    this.streetUpdated = this.streetUpdated.bind(this)
+    this.updatePositions = this.updatePositions.bind(this)
+  }
+
+  componentDidMount () {
+    window.addEventListener('stmx:set_street', this.streetUpdated)
+    window.addEventListener('stmx:width_updated', this.streetUpdated)
+    window.addEventListener('resize', this.updatePositions)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('stmx:set_street', this.streetUpdated)
+    window.removeEventListener('stmx:width_updated', this.streetUpdated)
+    window.removeEventListener('resize', this.updatePositions)
+  }
+
+  componentDidUpdate () {
+    this.updatePositions()
+  }
+
+  streetUpdated (e) {
+    const street = getStreet()
+    this.setState({street})
+  }
+
+  updatePositions () {
+    // TODO don't use the DOM when the whole app is react
+    const menuRect = document.querySelector('.menu-bar-right').getBoundingClientRect()
+
+    // TODO there might be a better way to do this than findDomNode
```

Comment:
The only way I've found to do this (and avoid using `findDOMNode`) is by creating a callback function passed as a prop to pick up refs from inside a component. Basically what this is doing in `Scrollable.jsx` (https://github.com/codeforamerica/streetmix/blob/master/assets/scripts/ui/Scrollable.jsx#L49-L52)